lms/services/document_url.py

```diff
@@ -93,17 +93,18 @@ def _from_assignment_in_db(self, request):
         for param in (
             "resource_link_id",  # A normal LTI (non-deep linked) launch
             "resource_link_id_history",  # A Blackboard course we can copy
-            "ext_d2l_resource_link_id_history",  # Ditto for Brightspace
+            "ext_d2l_resource_link_id_history",  # Ditto for D2L
+            "custom_ResourceLink.id.history",  # LTI 1.3 location for D2L and blackboard
```

Comment:
Our lti_parms will prefix all custom parameters with `custom_`

Note that:

- In D2L we were already including this but not using it
- In Blackboard the custom parameter was not configured.

Reply:
Ok, so this is when we're looking for the document URL in an `Assignment` stored in our DB, which we do if there's no deep linking-provided URL, Canvas files URL, or legacy VitalSource book URL. And this change means that if there's no `"resource_link_id"`, `"resource_link_id_history"` or `"ext_d2l_resource_link_id_history"` query param then we'll now look for a `"custom_ResourceLink.id.history"` query param as well, which we weren't looking for before. And in the case of D2L launches it is already sending us this `"custom_ResourceLink.id.history"` param, in the case of Blackboard admins will have to do an update to make it send us that param.

One thing I notice is that I think this means we'll potentially look for the `"custom_ResourceLink.id.history"` for *any* LTI launch (as long as all the earlier places where we look for a document URL don't find one). It is not restricted to only D2L and Blackboard LTI 1.3 launches, at least not explicitly. But that already seems to be the case with everything that `DocumentURLService` does, and I'm not sure that's a bad thing.